src/main/java/net/minecraftforge/fluids/FluidAttributes.java

```diff
@@ -363,12 +353,13 @@ public static Builder builder(String name, ResourceLocation stillTexture, Resour
         private int viscosity = 1000;
         private boolean isGaseous;
         private Rarity rarity = Rarity.COMMON;
+        private Function<Builder,FluidAttributes> factory;
 
-        protected Builder(String name, ResourceLocation stillTexture, ResourceLocation flowingTexture) {
-            this.name = name.toLowerCase(Locale.ENGLISH);
+        protected Builder(Fluid owner, ResourceLocation stillTexture, ResourceLocation flowingTexture, Function<Builder,FluidAttributes> factory) {
```

Comment:
As owner is only used for getting the registry name for the translation key, would it make sense to have two constructors, one that takes Fluid and one that takes a ResourceLocation (being the registry name). I don't fully remember the scope of how FluidAttributes gets made/used so this may not even make sense to do. (Or maybe it makes sense to change the protected builder to *only* ResourceLocation for registry name and make the `public static Builder builder(` above pass the registry name to this constructor.

Reply:
`FluidAttributes` construction is something that should only happen within the `createAttributes` method of the Fluid class, so in principle 100% of the times the given value in that parameter will be the `this`, within `createAttributes`. The main reason the builder isn't given privately to the createAttributes method directly, is that mod fluids might want to construct `FluidAttributes` instances with extra parameters not represented in the default attributes.